src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs

```diff
@@ -607,40 +604,37 @@ private class StatusBarProgress : RestoreOperationProgressUI
                 JoinableTaskFactory jtf,
                 CancellationToken token)
             {
-                return await jtf.RunAsync(async () =>
-                {
-                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                await jtf.SwitchToMainThreadAsync();
 
-                    var statusBar = serviceProvider.GetService<SVsStatusbar, IVsStatusbar>();
+                var statusBar = serviceProvider.GetService<SVsStatusbar, IVsStatusbar>();
 
-                    // Make sure the status bar is not frozen
-                    int frozen;
-                    statusBar.IsFrozen(out frozen);
+                // Make sure the status bar is not frozen
+                int frozen;
+                statusBar.IsFrozen(out frozen);
 
-                    if (frozen != 0)
-                    {
-                        statusBar.FreezeOutput(0);
-                    }
+                if (frozen != 0)
+                {
+                    statusBar.FreezeOutput(0);
+                }
 
-                    statusBar.Animation(1, ref icon);
+                statusBar.Animation(1, ref Icon);
```

Comment:
Thanks for the suggestion. We're planning to improve auto-restore UI in future NuGet/Home#4014.